app/src/main/java/com/example/perfectweatherallyear/utils/ForecastService.kt

```diff
@@ -0,0 +1,101 @@
+package com.example.perfectweatherallyear.utils
+
+import android.app.Notification
+import android.app.Service
+import android.content.Context
+import android.content.Intent
+import android.os.Build
+import android.os.Handler
+import android.os.IBinder
+import androidx.annotation.RequiresApi
+import com.example.perfectweatherallyear.appComponent
+import com.example.perfectweatherallyear.model.DayWeather
+import com.example.perfectweatherallyear.model.HourWeather
+import com.example.perfectweatherallyear.model.Location
+import com.example.perfectweatherallyear.repository.DataResult
+import com.example.perfectweatherallyear.repository.WeatherRepository
+import com.example.perfectweatherallyear.ui.weekWeather.DAYS_NUMBER
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import java.text.SimpleDateFormat
+import java.util.*
+import javax.inject.Inject
+
+class ForecastService : Service() {
+    @Inject
+    lateinit var repository: WeatherRepository
+    lateinit var dayWeatherForNotification: DayWeather
+    lateinit var hourWeatherForNotification: HourWeather
+    private lateinit var mHandler: Handler
+    private lateinit var mRunnable: Runnable
+
+    override fun attachBaseContext(newBase: Context?) {
+        super.attachBaseContext(newBase)
+        this.appComponent.inject(this)
+    }
+
+    override fun onBind(intent: Intent): IBinder? {
+        throw UnsupportedOperationException("Not yet implemented")
+    }
+
+    override fun onCreate() {
+        super.onCreate()
+        mHandler = Handler()
+        var notification = Notification()
+        mRunnable = Runnable{
+            GlobalScope.launch {
+                notification = NotificationHandler.createNotification(
```

Comment:
You are trying to do some work in onCreate. But you should use onStartCommand for implementing main work of the service.